setup.py

```diff
@@ -34,7 +35,7 @@
 ]
 # Define all extras
 extras = {
-    "code_style": ["flake8<3.8.0,>=3.7.0", "black", "pre-commit==1.17.0"],
+    "code_style": ["pre-commit~=2.12"],
```

Comment:
does pre-commit handle all the installs for us now?

Reply:
Pre-commit installs all "hooks" (including black, flake8, and the othes defined in the `.pre-commit-config.yaml`) into separate virtual environments. This has always actually been the case